src/include/sof/audio/buffer.h

```diff
@@ -295,4 +295,27 @@ static inline int buffer_set_params(struct comp_buffer *buffer,
 	return 0;
 }
 
+static inline bool buffer_params_match(struct comp_buffer *buffer,
+				       struct sof_ipc_stream_params *params,
+				       uint32_t flag)
+{
+	bool matched = true;
+
+	assert(params && buffer);
+
+	if (flag & BUFF_PARAMS_FRAME_FMT)
+		matched = matched &&
+			  (buffer->stream.frame_fmt == params->frame_fmt);
+
+	if (flag & BUFF_PARAMS_RATE)
+		matched = matched &&
+			  (buffer->stream.rate == params->rate);
+
+	if (flag & BUFF_PARAMS_CHANNELS)
+		matched = matched &&
+			  (buffer->stream.channels == params->channels);
+
```

Comment:
There is no point in doing the subsequent check if the first one fails.
```c
if ((flag & FRAME_FMT) && (buffer->forrmat != params->format))
    return false;
/* same for rate and channels */
```

Reply:
> There is no point in doing the subsequent check if the first one fails.
> 
> ```c
> if ((flag & FRAME_FMT) && (buffer->forrmat != params->format))
>     return false;
> /* same for rate and channels */
> ```

Correct, will change it.